### DISCUSSION TITLE
Cap the max falling speed of an object

### DISCUSSION THREAD

**Discussion by ia97lies** [dyn4j/dyn4j#284] [Q&A]
Hi, is there a good way to cap the speed of an object in freefall? I need this to cap the falling speed in a jump to make it more predictable. Something like Celeste does as well.

**Comment by wnbittle** [comment]
You want the linear velocity in the -y direction to be capped?

One way you could achieve this would be to override the `Body.integratePosition` method.  Before calling `super.integratePosition(...)` you can check the velocity and adjust it then.  

There's some danger in this since the velocity was potentially computed from a number of sources (forces, gravity, joints, contacts, etc).  The joints and contacts in particular use velocity to enforce their respective behavior.

**Reply by ia97lies** [reply]
Thank you! Sounds quite complicated... I found a simpler way, I just set the linear velocity back to the cap value as soon it is higher on every update.
